src/Security/Authentication/Core/src/RemoteAuthenticationHandler.cs

```diff
@@ -156,6 +191,8 @@ public virtual async Task<bool> HandleRequestAsync()
         /// </summary>
         protected abstract Task<HandleRequestResult> HandleRemoteAuthenticateAsync();
 
+        protected abstract IEnumerable<Endpoint> GetEndpoints();
```

Comment:
Same with this one, edpoints need to be resolved during startup, not from per-request handlers.